server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1044,7 +1044,12 @@ public void performOn(
             }
             final ConcreteReplicaRequest<ReplicaRequest> replicaRequest = new ConcreteReplicaRequest<>(
                 request, replica.allocationId().getId(), primaryTerm, globalCheckpoint, maxSeqNoOfUpdatesOrDeletes);
-            sendReplicaRequest(replicaRequest, node, listener);
+            final ActionListenerResponseHandler<ReplicaResponse> handler = new ActionListenerResponseHandler<>(listener, in -> {
+                ReplicaResponse replicaResponse = new ReplicaResponse();
+                replicaResponse.readFrom(in);
```

Comment:
perhaps, while you're at it, implement the StreamInput constructor for ReplicaResponse and override readFrom by throwing exception, just like `AddVotingConfigExclusionsResponse` does. This can then just be `new ActionListenerResponseHandler<>(listener, ReplicaResponse::new)`

Reply:
The `Writeable` implementations are on their way as a followup.